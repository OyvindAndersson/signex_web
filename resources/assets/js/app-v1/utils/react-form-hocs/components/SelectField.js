import React from 'react'
import Select from 'react-select'
import { Form, Text, Field, withFieldApi} from 'react-form'
/**
 * Select Wrapper
 * 
 * react-form HOC of the react-select component
 * use "props.items" to set data for the select
 * component. Read react-select and react-form
 * API for general use.
 */

export default class SelectField extends React.Component {
    componentDidCatch(error, info){
        console.log("FUCKING FAIL")
        console.error(error)
        console.debug(info)
    }
    validate() {

    }
    render() {
        const { field } = this.props || 'selectField'
        const defaultValue = this.props.value ? this.props.value : 0

        return(
            <Field validate={this.validate} field={field}>
                { fieldApi => {
                    const { onChange, onBlur, field, options, ...rest } = this.props
                    const { value, error, warning, success, setValue, setTouched } = fieldApi
                    return(
                    <Select 
                        value={value || '' }
                        onChange={(e) => {
                            setValue(e ? e.value : e)
                            if(onChange){
                                const id = this.props.id ? this.props.id : null
                                onChange(e, id)
                            }
                        }}
                        options={options}
                        {...rest} />
                    )
                }}
            </Field>
        )
    }
}