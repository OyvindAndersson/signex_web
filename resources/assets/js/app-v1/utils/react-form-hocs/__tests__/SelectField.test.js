import React, {Component} from 'react'
import renderer from 'react-test-renderer'

import { Form } from 'react-form'
import SelectField from '../components/SelectField'

describe('SelecField component', () => {

    it('Should render', () => {
        const options = [{}]
        let tree = renderer.create(
            <Form >
            { formApi => (
                <form onSubmit={() => {}} id="form5">
                    <SelectField required={true}
                                field="order.client_id" 
                                id="clientInput" 
                                options={options}
                                value={1}>
                    </SelectField>
                </form>
            )}
            </Form>
            
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })

})
