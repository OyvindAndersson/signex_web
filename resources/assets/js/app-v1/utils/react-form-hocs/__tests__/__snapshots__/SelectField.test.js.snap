// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelecField component Should render 1`] = `
<form
  id="form5"
  onSubmit={[Function]}
>
  <div
    className="Select is-clearable is-searchable Select--single"
    style={undefined}
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-placeholder"
        >
          Select...
        </div>
        <div
          className="Select-input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant="react-select-2--value"
            aria-describedby={undefined}
            aria-expanded="false"
            aria-haspopup="false"
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className={undefined}
            id="clientInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            role="combobox"
            style={
              Object {
                "boxSizing": "content-box",
                "width": "5px",
              }
            }
            tabIndex={undefined}
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</form>
`;
